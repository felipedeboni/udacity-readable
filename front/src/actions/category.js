export const FETCH_CATEGORIES = 'FETCH_CATEGORIES';
export const FETCH_CATEGORIES_SUCCESS = 'FETCH_CATEGORIES_SUCCESS';
export const FETCH_CATEGORIES_FAIL = 'FETCH_CATEGORIES_FAIL';

export const fetchCategories = () => {
  return {
    type: FETCH_CATEGORIES,
    payload: {
      request: {
        url: '/categories'
      }
    }
  };
};
