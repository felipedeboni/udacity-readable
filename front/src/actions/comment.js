import uuid from 'utils/uuid';

// GET /posts/:id/comments
// Get all the comments for a single post.
export const FETCH_COMMENTS_BY_POST = 'FETCH_COMMENTS_BY_POST';
export const FETCH_COMMENTS_BY_POST_SUCCESS = 'FETCH_COMMENTS_BY_POST_SUCCESS';
export const FETCH_COMMENTS_BY_POST_FAIL = 'FETCH_COMMENTS_BY_POST_FAIL';
export const fetchCommentsByPost = postId => {
  return {
    type: FETCH_COMMENTS_BY_POST,
    payload: {
      request: {
        url: `/posts/${postId}/comments`
      }
    }
  };
};

// GET /comments/:id
// Get the details for a single comment.
export const FETCH_COMMENT = 'FETCH_COMMENT';
export const FETCH_COMMENT_SUCCESS = 'FETCH_COMMENT_SUCCESS';
export const FETCH_COMMENT_FAIL = 'FETCH_COMMENT_FAIL';
export const fetchComment = id => {
  return {
    type: FETCH_COMMENT,
    payload: {
      request: {
        url: `/comments/${id}`
      }
    }
  };
};

// POST /comments
// Add a comment to a post.
export const ADD_COMMENT = 'ADD_COMMENT';
export const ADD_COMMENT_SUCCESS = 'ADD_COMMENT_SUCCESS';
export const ADD_COMMENT_FAIL = 'ADD_COMMENT_FAIL';
export const addComment = (parentId, values) => {
  const { author, body } = values;
  const data = {
    id: uuid(),
    timestamp: Date.now(),
    parentId,
    author,
    body
  };

  return {
    type: ADD_COMMENT,
    payload: {
      request: {
        method: 'POST',
        url: '/comments',
        data
      }
    }
  };
};

// PUT /comments/:id
// Edit the details of an existing comment.
export const EDIT_COMMENT = 'EDIT_COMMENT';
export const EDIT_COMMENT_SUCCESS = 'EDIT_COMMENT_SUCCESS';
export const EDIT_COMMENT_FAIL = 'EDIT_COMMENT_FAIL';
export const editComment = (id, data) => {
  return {
    type: EDIT_COMMENT,
    payload: {
      request: {
        method: 'PUT',
        url: `/comments/${id}`,
        data
      }
    }
  };
};

// DELETE /comments/:id
// Sets a comment's deleted flag to true.
export const REMOVE_COMMENT = 'REMOVE_COMMENT';
export const REMOVE_COMMENT_SUCCESS = 'REMOVE_COMMENT_SUCCESS';
export const REMOVE_COMMENT_FAIL = 'REMOVE_COMMENT_FAIL';
export const removeComment = id => {
  return {
    type: REMOVE_COMMENT,
    payload: {
      request: {
        method: 'DELETE',
        url: `/comments/${id}`
      }
    }
  };
};

// POST /comments/:id
// Used for voting on a comment.
export const VOTE_COMMENT = 'VOTE_COMMENT';
export const VOTE_COMMENT_SUCCESS = 'VOTE_COMMENT_SUCCESS';
export const VOTE_COMMENT_FAIL = 'VOTE_COMMENT_FAIL';
const vote = (id, type) => {
  return {
    type: VOTE_COMMENT,
    payload: {
      request: {
        method: 'POST',
        url: `/comments/${id}`,
        data: {
          option: type
        }
      }
    }
  };
};

export const voteCommentUp = id => {
  return vote(id, 'upVote');
};

export const voteCommentDown = id => {
  return vote(id, 'downVote');
};
