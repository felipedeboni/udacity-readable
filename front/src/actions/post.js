import uuid from 'utils/uuid';

// GET /posts
// Get all of the posts. Useful for the main page when no category is selected.
export const FETCH_ALL_POSTS = 'FETCH_ALL_POSTS';
export const FETCH_ALL_POSTS_SUCCESS = 'FETCH_ALL_POSTS_SUCCESS';
export const FETCH_ALL_POSTS_FAIL = 'FETCH_ALL_POSTS_FAIL';
export const fetchAllPosts = () => {
  return {
    type: FETCH_ALL_POSTS,
    payload: {
      request: {
        url: `/posts`
      }
    }
  };
};

// GET /:category/posts
// Get all of the posts for a particular category.
export const FETCH_POSTS_BY_CATEGORY = 'FETCH_POSTS_BY_CATEGORY';
export const FETCH_POSTS_BY_CATEGORY_SUCCESS =
  'FETCH_POSTS_BY_CATEGORY_SUCCESS';
export const FETCH_POSTS_BY_CATEGORY_FAIL = 'FETCH_POSTS_BY_CATEGORY_FAIL';
export const fetchPostsByCategory = categoryId => {
  return {
    type: FETCH_POSTS_BY_CATEGORY,
    payload: {
      request: {
        url: `/${categoryId}/posts`
      }
    }
  };
};

// GET /posts/:id
// Get the details of a single post.
export const FETCH_POST = 'FETCH_POST';
export const FETCH_POST_SUCCESS = 'FETCH_POST_SUCCESS';
export const FETCH_POST_FAIL = 'FETCH_POST_FAIL';
export const fetchPost = id => {
  return {
    type: FETCH_POST,
    payload: {
      request: {
        url: `/posts/${id}`
      }
    }
  };
};

// POST /posts
// Add a new post.
export const ADD_POST = 'ADD_POST';
export const ADD_POST_SUCCESS = 'ADD_POST_SUCCESS';
export const ADD_POST_FAIL = 'ADD_POST_FAIL';
export const addPost = ({ category, author, title, body }) => {
  const data = {
    id: uuid(),
    timestamp: Date.now(),
    category,
    author,
    title,
    body
  };

  return {
    type: ADD_POST,
    payload: {
      request: {
        method: 'POST',
        url: '/posts',
        data
      }
    }
  };
};

// PUT /posts/:id
// Edit the details of an existing post.
export const EDIT_POST = 'EDIT_POST';
export const EDIT_POST_SUCCESS = 'EDIT_POST_SUCCESS';
export const EDIT_POST_FAIL = 'EDIT_POST_FAIL';
export const editPost = (id, data) => {
  return {
    type: EDIT_POST,
    payload: {
      request: {
        method: 'PUT',
        url: `/posts/${id}`,
        data
      }
    }
  };
};

// DELETE /posts/:id
// Sets the deleted flag for a post to 'true'.
// Sets the parentDeleted flag for all child comments to 'true'.
export const REMOVE_POST = 'REMOVE_POST';
export const REMOVE_POST_SUCCESS = 'REMOVE_POST_SUCCESS';
export const REMOVE_POST_FAIL = 'REMOVE_POST_FAIL';
export const removePost = id => {
  return {
    type: REMOVE_POST,
    payload: {
      id: id,
      request: {
        method: 'DELETE',
        url: `/posts/${id}`
      }
    }
  };
};

// POST /posts/:id
// Used for voting on a post.
export const VOTE_POST = 'VOTE_POST';
export const VOTE_POST_SUCCESS = 'VOTE_POST_SUCCESS';
export const VOTE_POST_FAIL = 'VOTE_POST_FAIL';
const vote = (id, type) => {
  return {
    type: VOTE_POST,
    payload: {
      request: {
        method: 'POST',
        url: `/posts/${id}`,
        data: {
          option: type
        }
      }
    }
  };
};

export const votePostUp = id => {
  return vote(id, 'upVote');
};

export const votePostDown = id => {
  return vote(id, 'downVote');
};
