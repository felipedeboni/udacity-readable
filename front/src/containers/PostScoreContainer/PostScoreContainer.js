import React from 'react';
import { connect } from 'react-redux';
import { votePostUp, votePostDown } from 'actions/post';
import { Score } from 'components';

const PostScoreContainer = ({ id, score, voteUp, voteDown }) => {
  return (
    <Score
      id={id}
      score={score}
      onVoteUp={voteUp}
      onVoteDown={voteDown}
    />
  );
}

const mapProps = (state, ownProps) => {
  const post = state.posts[ownProps.id];
  return {
    score: post.voteScore
  };
};

const mapDispatch = dispatch => {
  return {
    voteUp: postId => dispatch(votePostUp(postId)),
    voteDown: postId => dispatch(votePostDown(postId))
  };
};

export default connect(mapProps, mapDispatch)(PostScoreContainer);
