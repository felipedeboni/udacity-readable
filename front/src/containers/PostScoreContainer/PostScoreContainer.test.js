import React from 'react';
import * as PostActions from 'actions/post';
import PostScoreContainer from './PostScoreContainer';

describe('PostScoreContainer', () => {
  const render = () => {
    const { postsObject } = global.mock;
    const [post] = global.mock.posts;
    const state = {
      posts: postsObject
    };

    const { component, store } = global.shallowWithStore(
      <PostScoreContainer id={post.id} />,
      state
    );

    return {
      component,
      store,
      post
    };
  };

  it('renders without crashing', () => {
    const { component } = render();

    expect(component.dive()).toMatchSnapshot();
  });

  it('renders the Score component correctly', () => {
    const { component, post } = render();
    const score = component.dive().dive();

    expect(
      score
        .find('.score__number')
        .text()
        .trim()
    ).toBe(post.voteScore.toString());
  });

  it('dispatch voteUp', () => {
    const { component, store } = render();
    const score = component.dive().dive();

    score.find('.score__vote--up').simulate('click');
    expect(
      store.getActions().find(_ => _.type === PostActions.VOTE_POST)
    ).toBeDefined();
  });

  it('dispatch voteDown', () => {
    const { component, store } = render();
    const score = component.dive().dive();

    score.find('.score__vote--down').simulate('click');
    expect(
      store.getActions().find(_ => _.type === PostActions.VOTE_POST)
    ).toBeDefined();
  });
});
