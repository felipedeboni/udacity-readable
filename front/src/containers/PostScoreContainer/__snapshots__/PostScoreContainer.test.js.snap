// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostScoreContainer renders without crashing 1`] = `
<Score
  id="8xf0y6ziyjabvozdd253nd"
  onVoteDown={[Function]}
  onVoteUp={[Function]}
  score={6}
/>
`;
